### ISSUE TITLE
Issue coloring two edges differently when output from two different anchors on the same node

### ISSUE THREAD

**Issue by eliasdefaria** [open-source-labs/Svelvet#327]
Hi there, big fan of Svelvet so thanks for your work on this! I've run into a small edge coloring issue, here's some context:

I have a lineage node defined as a custom component like this:
```svelte title="LineageNode.svelte"
<Anchor input>
    <div class="bg-transparent" />
</Anchor>
<button 
    on:click={handleNodeSelect} 
    style={`background-color: ${node.data.bgColor}; border-color: ${node.data.borderColor};`}
    class="w-full h-full cursor-pointer border rounded-lg hover:!border-secondary-turquoise hover:shadow-md hover:shadow-secondary-turquoise-super-light node-transitions"
    use:tooltip={{ content: node.data.target, placement: 'top', delay: 300 }}
>
    <span class="font-medium w-40 text-ellipsis inline-block whitespace-nowrap overflow-hidden">{node.data.label}</span>
</button>  
{ #each connections as connection }
    <Anchor output edgeColor={connectionColorMap[connection]} edge={LineageEdge}>
        <div class="bg-transparent" />
    </Anchor>
{/each}
```

The color of the edge comes from a map of writable stores for the `edgeColor` property on the output `<Anchor>` like so:

```typescript
let connections = safeParseJSON<string[]>(JSON.stringify(node.connections ?? [])) ?? [];
const connectionColorMap = connections.reduce((acc, connection) => {
        acc[Array.isArray(connection) ? connection[0] : connection] = writable(getEdgeColor(nodes, Array.isArray(connection) ? connection[0] : connection, node.id as string));
        return acc;
}, {} as Record<string, Writable<CSSColorString>>);
```

Lastly, I subscribe to changes on a store and set the color of the edge accordingly:
```typescript
lineageStore.subscribe((_) => {
        Object.entries(connectionColorMap).forEach(([ connection, edgeColorStore ]) => {
            edgeColorStore.set(getEdgeColor(nodes, connection, node.id));
        });
});
```

This works for most cases, but the coloring breaks when one downstream node has more than one upstream node (see photo below). My goal in the photo below is to have the edge from `main` -> `five` be colored differently from the edge from `main` -> `one`. I can confirm the store value of each of the colors is updating accordingly, so I've now come to assume this may be a bug on Svelvet's end? Either that or I'm misusing the `<Anchor>` functionality. 

Thanks for your help!

<img width="511" alt="Screenshot 2023-05-04 at 1 24 50 PM" src="https://user-images.githubusercontent.com/30669330/236323716-097f5bb3-e3ba-42f0-9b20-311066f3f8c8.png">

**Comment by briangregoryholmes** [comment]
Hi, Elias. I'm trying to reproduce this, but I think there are some implementation details on your end that I'm not able to recreate, so I haven't been able to pin down what issue you might be running into. It looks to me like you're creating multiple invisible anchors (which there is a prop for now, actually!) and then stacking them on top of each other. We definitely plan on implementing this idea natively into a single Anchor, but I see no reason why the general idea of what you're doing shouldn't work.

This is what I threw together on trying to reproduce. 

```html

<script lang="ts">
	import { Node, Anchor, type CSSColorString } from 'svelvet';
	import { writable } from 'svelte/store';
	import type { Writable } from 'svelte/store';

	const connections: Record<string, Writable<CSSColorString>> = {
		edgeOne: writable('blue'),
		edgeTwo: writable('green')
	};

	const changeOneEdgeRandomly = () => {
		const colors: Array<CSSColorString> = ['blue', 'green', 'yellow', 'red', 'black', 'white'];
		const edge = Math.random() > 0.5 ? 'edgeOne' : 'edgeTwo';
		const color = colors[Math.floor(Math.random() * colors.length)];
		connections[edge].set(color);
		setTimeout(() => changeOneEdgeRandomly(), 1000);
	};

	setTimeout(changeOneEdgeRandomly, 1000);
</script>

<Node useDefaults dimensions={{ width: 200, height: 200 }}>
	{#each Object.entries(connections) as [key, store] (key)}
		<Anchor edgeColor={connections[key]} />
	{/each}
</Node>
```

**Comment by eliasdefaria** [comment]
Hi Brian! Thanks for getting back to me. Yep, that's exactly what's going on. I have two transparent anchors stacked on top of each other. Your example looks pretty similar to what I'm doing, let me see if I can put together a Sandbox to try and reproduce.

**Comment by eliasdefaria** [comment]
Hey Brian, here's a link to a Sandbox reproducing the issue: https://stackblitz.com/edit/svelvet-v7-lzc8pw?file=src/routes/+page.svelte

I tried to replicate the structure of my project as best as possible, but removed anything unnecessary / identifying for the project (hence things may seem a bit randomly structured).

To repro:

1. Note that all edges currently start green. This is expected
2. Click the button in the center of any of the nodes. This will set the lineageStore to the value ['main', 'one', 'two']. Note that it is setting the value to an array excluding the node with id 'five'
3. All edges should then turn red. The expected behavior is that only the edges between 'main', 'one', and 'two' turn red. Not the edge between 'main' and 'five'. This behavior should happen since `getEdgeColor` checks to see if `lineageStore` contains both of the nodes that it is connecting. Since 'five' is not present in `lineageStore`, `getEdgeColor` returns `green` yet `red` is still rendered.

Let me know if you have any questions. Thanks so much for your help!

P.S: One other bug I've encountered in this project (can be reproduced in the sandbox as well) is that the edges disappear on hot reload from vite. I have to fully reload the page to get them to re-render.

**Comment by briangregoryholmes** [comment]
Found the issue. It's in a completely unrelated place and is just related to our connection processing logic when creating custom Nodes. I'll have it fixed tonight.

Essentially, both Edges from main are connected to that first Anchor. This was obfuscated by the fact that they were invisible and stacked on top of each other. You could set outputs={2} on the wrapping Node as a temporary workaround, but let me make some adjustments to the code.

**Comment by eliasdefaria** [comment]
Awesome! Thanks for hopping on it so quick. I'll wait for the next release

**Comment by briangregoryholmes** [comment]
Fix is live in v7.0.41. Thanks for bringing this to our attention and let me know if it behaves how you expect now. It at least works for your sandbox example and some updated tests on our end.
